x/ccv/provider/keeper/relay.go

```diff
@@ -343,10 +343,31 @@ func (k Keeper) OnRecvSlashPacket(
 		return ccv.V1Result, nil
 	}
 
-	// Check that chain is launched and the validator belongs to the consumer chain valset
-	if k.GetConsumerPhase(ctx, consumerId) == providertypes.CONSUMER_PHASE_LAUNCHED && !k.IsConsumerValidator(ctx, consumerId, providerConsAddr) {
-		k.Logger(ctx).Error("cannot jail validator %s that does not belong to consumer %s valset",
-			providerConsAddr.String(), consumerId)
+	// check that the chain is launched
+	if k.GetConsumerPhase(ctx, consumerId) != providertypes.CONSUMER_PHASE_LAUNCHED {
+		k.Logger(ctx).Error("cannot jail validator on a chain that is not currently launched",
+			"consumerId", consumerId,
+			"provider cons addr", providerConsAddr.String(),
+		)
+
+		// Naturally, we do not return a slash ack here because the fact that we are here means:
+		// i)  the chain was at some point launched and hence `k.GetChannelIdToConsumerId` found a consumer (at the beginning
+		//     of the `OnRecvSlashPacket` method);
+		// ii) the chain is not launched anymore which means the chain is now stopped (with `MsgRemoveConsumer`),
+		//     but the chain is not yet fully removed because otherwise `k.GetChannelIdToConsumerId` would have not found
+		//     a consumer.
+		// Therefore, we should not expect any more packets to be sent from this chain.
```

Comment:
I'm not sure I agree with this argument. The fact that the consumer is stopped on the provider, it doesn't mean that it's nodes are down. The consumer can still make progress and send SlashPackets to the provider.